CI: cuFFT callback tests failed due to `libcufft_static.a` not found

For example, this innocent PR https://github.com/cupy/cupy/pull/10129 hit cufft callback errors:
```
E           AssertionError: Only cupy raises error
E
E           Traceback (most recent call last):
E             File "/root/.local/lib/python3.10/site-packages/cupy/testing/_loops.py", line 54, in _call_func
E               result = impl(*args, **kw)
E             File "/tmp/tmp.IuRY4WRBZK/tests/cupy_tests/fft_tests/test_callback.py", line 568, in test_fft_load_aux
E               with xp.fft.config.set_cufft_callbacks(
E             File "cupy/fft/_callback.pyx", line 833, in cupy.fft._callback.set_cufft_callbacks.__init__
E             File "cupy/fft/_callback.pyx", line 437, in cupy.fft._callback._LegacyCallbackManager.__init__
E             File "cupy/fft/_callback.pyx", line 242, in cupy.fft._callback._prune
E             File "/opt/pyenv/versions/3.10.18/lib/python3.10/subprocess.py", line 457, in check_returncode
E               raise CalledProcessError(self.returncode, self.args, self.stdout,
E           subprocess.CalledProcessError: Command '['/usr/local/cuda/bin/nvprune', '--generate-code=arch=compute_70,code=sm_70', '--generate-code=arch=compute_72,code=sm_72', '--generate-code=arch=compute_75,code=sm_75', '/usr/local/cuda-12.0/targets/x86_64-linux/lib64/libcufft_static.a', '-o', '/tmp/cupy_cache/.cupy/kernel_cache/tmpw93e5klb/libcufft_static_11001_sm7.a']' returned non-zero exit status 1.

self       = CompletedProcess(args=['/usr/local/cuda/bin/nvprune', '--generate-code=arch=compute_70,code=sm_70', '--generate-code=a...r=b"nvprune fatal   : Could not open input file '/usr/local/cuda-12.0/targets/x86_64-linux/lib64/libcufft_static.a'\n")
```
This happens on both main and v14 branches starting today (or yesterday). Here are two example failing pipelines:
- https://ci.preferred.jp/cupy.linux.cuda120/223030/
- https://ci.preferred.jp/cupy.linux.cuda129/223039/

--- a/cupy/fft/_callback.pyx
+++ b/cupy/fft/_callback.pyx
@@ -223,9 +223,26 @@ cdef dict _cc_major_map = {
 
 cdef inline str _prune(str temp_dir, str cache_dir, str _cufft_ver, str arch):
     cdef str cufft_lib_full, cufft_lib_pruned, cufft_lib_temp, cufft_lib_cached
+    cdef str libdir, candidate
 
     if _nvprune:
-        cufft_lib_full = os.path.join(_cuda_path, 'lib64/libcufft_static.a')
+        # _cuda_path may point to either the CUDA Toolkit root (which has
+        # lib64/ on Linux) or a targets/<arch> directory (which has lib/
+        # only; e.g. conda, or a system CTK located via cuda.pathfinder),
+        # so probe both layouts.
+        cufft_lib_full = None
+        for libdir in ('lib64', 'lib'):
+            candidate = os.path.join(_cuda_path, libdir, 'libcufft_static.a')
+            if os.path.isfile(candidate):
+                cufft_lib_full = candidate
+                break
+        if cufft_lib_full is None:
+            # cannot locate the static library; fall back to linking the
+            # full one via nvcc's default library search paths
+            warnings.warn(
+                'libcufft_static.a not found under ' + _cuda_path,
+                RuntimeWarning)
+            return None
         cufft_lib_pruned = f'cufft_static_{_cufft_ver}_sm{arch[0]}'
         cufft_lib_temp = os.path.join(temp_dir,
                                       'lib' + cufft_lib_pruned + '.a')

